### PR TITLE
[solution] uk/medium-capitalize

### PR DESCRIPTION
**Pull request by shalamai**
# Challenge [medium-capitalize]

## Checklist

- [ ] Make sure that you are opening the PR with a single solution (one solution per PR)
- [ ] The solution must have at least several references for the readers to dive in depth

### CODE REVIEW

**Comment by ghaiklor** [comment]
@bdvorianov bump

File: uk/medium-capitalize.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+```typescript
+type capitalized = Capitalize<'hello world'> // expected to be 'Hello world'
+```
+
+## Розв'язок
+
+Спочатку, я не зрозумів завдання.
+Ми не можемо реалізувати загальний розв'язок для того, щоб зробити літеру великою.
+Тому, TypeScript компілятор надає вбудований тип `Capitalize` (вбудований), за допомогою якого ця проблема вирішується досить легко:
```
Comment (bdvorianov) [inline uk/medium-capitalize.md:22]:
```suggestion
Тому, TypeScript компілятор надає вбудований тип `Capitalize`, за допомогою якого ця проблема вирішується досить легко:
```

File: uk/medium-capitalize.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+```typescript
+type MyCapitalize<S extends string> = Capitalize<S>
+```
+
+Але я сумніваюся, що намір автора був саме в цьому.
```
Comment (bdvorianov) [inline uk/medium-capitalize.md:28]:
```suggestion
Але я сумніваюся, що задум автора був саме в цьому.
```

File: uk/medium-capitalize.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+Є словник з великими літерами.
+Давайте тепер виведемо першу літеру із рядкового літерала.
+Для цього, скористаємося класичною інструкцією з умовним типом.
+
+```typescript
+type Capitalize<S> = S extends `${infer C}${infer T}` ? C : S;
+```
+
+В тип параметрі `C` отримуємо першу літеру, а в `T` - решту рядка.
```
Comment (bdvorianov) [inline uk/medium-capitalize.md:47]:
```suggestion
В тип-параметрі `C` отримуємо першу літеру, а в `T` - решту рядка.
```

File: uk/medium-capitalize.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+Перевіримо чи існує перша літера в нашому словнику.
```
Comment (bdvorianov) [inline uk/medium-capitalize.md:48]:
```suggestion
Перевіримо, чи існує перша літера в нашому словнику.
```

**Comment by bdvorianov** [comment]
Кілька помарочок, а в іншому, як кажуть, LGTM 👌🏻

**Review by bdvorianov — APPROVED** [review]

**Pull request merged**